src_js/index.js

```diff
@@ -1,17 +1,18 @@
-'use strict';
-
 var os = require('os');
 
-// require and export only for platforms where the module is supported.
-// Individual module require'ed use features of node.js that would trigger
-// syntax errors in older versions. This allows for the module to be included
-// in applications like Tedious which supports older version of node.js, even
-// if the functionality itself won't be available. The application can decide
-// what to do if the module is not supported on the platform where it's running.
-if (require('semver').gte(process.version, '4.0.0') && os.type() === 'Windows_NT') {
-  module.exports.ModuleSupported = true;
+const NativeAuthProvider = require('./native.js').NativeAuthProvider;
 
-  module.exports.SspiClientApi = require('./sspi_client');
-  module.exports.Fqdn = require('./fqdn');
-  module.exports.MakeSpn = require('./make_spn');
-}
+module.exports = function(options) {
+  // require and export only for platforms where the module is supported.
+  // Individual module require'ed use features of node.js that would trigger
+  // syntax errors in older versions. This allows for the module to be included
+  // in applications like Tedious which supports older version of node.js, even
+  // if the functionality itself won't be available. The application can decide
+  // what to do if the module is not supported on the platform where it's running.
+  if (require('semver').gte(process.version, '4.0.0') && os.type() === 'Windows_NT') {
```

Comment:
What do you think about putting the version requirement for Node.js into the `package.json` file, and removing it from here? That'd allow you to get rid of the `semver` dependency as well.